### PR TITLE
[query] Fix /m3query returning 500 on invalid query argument

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Fixes `m3query/api/v1/{query,query_range}` returning 500 on invalid query string. Plus additional tests

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/tools/dtest/docker/harness/query_api_test.go [hunk 1]
```diff
@@ -38,12 +38,14 @@ type urlTest struct {
 func TestInvalidInstantQueryReturns400(t *testing.T) {
 	coord := singleDBNodeDockerResources.Coordinator()
 
-	instantQueryBadRequestTest := []urlTest{
+	urlPrefix := []string{"", "prometheus/", "m3query/"}
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/query_api_test.go:41]: Nit: make the name plural.

File: src/cmd/tools/dtest/docker/harness/query_api_test.go [hunk 2]
```diff
@@ -38,12 +38,14 @@ type urlTest struct {
...
+
+	instantQueryBadRequestTest := addPrefixes(urlPrefix, []urlTest{
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/query_api_test.go:43]: Nit: make the name plural.

File: src/cmd/tools/dtest/docker/harness/query_api_test.go [hunk 3]
```diff
@@ -55,7 +57,9 @@ func TestInvalidInstantQueryReturns400(t *testing.T) {
 func TestInvalidRangeQueryReturns400(t *testing.T) {
 	coord := singleDBNodeDockerResources.Coordinator()
 
-	queryBadRequestTest := []urlTest{
+	urlPrefix := []string{"", "prometheus/", "m3query/"}
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/query_api_test.go:60]: Nit: make the name plural.

File: src/cmd/tools/dtest/docker/harness/query_api_test.go [hunk 4]
```diff
@@ -55,7 +57,9 @@ func TestInvalidInstantQueryReturns400(t *testing.T) {
...
+
+	queryBadRequestTest := addPrefixes(urlPrefix, []urlTest{
```
Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/query_api_test.go:62]: Nit: make the name plural.

Comment (linasm) [inline src/cmd/tools/dtest/docker/harness/query_api_test.go:62]: I think these two tests only differ by their test table contents and so they can be safely merged. Having two separate test methods made me try figuring out what is the principal difference between them.

**Review by linasm — CHANGES_REQUESTED** [review]

File: src/cmd/tools/dtest/docker/harness/query_api_test.go (hunk 1 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/query_api_test.go]: https://github.com/m3db/m3/pull/2916/commits/661c9ae35af62bd81232983341df94f762e1f6c0#diff-391a291f1f91098bc731847f8843207e5327fe87a52a4599a45ac0cf4c5732e7R39

File: src/cmd/tools/dtest/docker/harness/query_api_test.go (hunk 2 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/query_api_test.go]: https://github.com/m3db/m3/pull/2916/commits/661c9ae35af62bd81232983341df94f762e1f6c0#diff-391a291f1f91098bc731847f8843207e5327fe87a52a4599a45ac0cf4c5732e7R41

File: src/cmd/tools/dtest/docker/harness/query_api_test.go (hunk 3 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/query_api_test.go]: https://github.com/m3db/m3/pull/2916/commits/661c9ae35af62bd81232983341df94f762e1f6c0#diff-391a291f1f91098bc731847f8843207e5327fe87a52a4599a45ac0cf4c5732e7R52

File: src/cmd/tools/dtest/docker/harness/query_api_test.go (hunk 4 above)
Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/query_api_test.go]: https://github.com/m3db/m3/pull/2916/commits/661c9ae35af62bd81232983341df94f762e1f6c0#diff-391a291f1f91098bc731847f8843207e5327fe87a52a4599a45ac0cf4c5732e7R54

Comment (vpranckaitis) [reply src/cmd/tools/dtest/docker/harness/query_api_test.go]: https://github.com/m3db/m3/pull/2916/commits/661c9ae35af62bd81232983341df94f762e1f6c0#diff-391a291f1f91098bc731847f8843207e5327fe87a52a4599a45ac0cf4c5732e7R69-R75

**Review by linasm — APPROVED** [review]
LGTM

**Review by soundvibe — APPROVED** [review]
LGTM

**Pull request merged**